samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -188,6 +189,7 @@ void doOnProcessorChange(List<String> processors) {
 
     // Generate the JobModel
     JobModel jobModel = generateNewJobModel(currentProcessorIds);
+    JobModelManager.createChangeLogStreams(new StorageConfig(config), jobModel.maxChangeLogStreamPartitions);
```

Comment:
Where is the maxChangeLogStreamPartitions being set on the JobModel?

Reply:
Set in JobModel constructor.

It's populated in JobModelManager.readJobModel(which generates changeLogPartitionId of the tasks) and JobModel constructor sets it to maximum changeLogPartitionId of all tasks.